src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java

```diff
@@ -156,16 +156,38 @@ public synchronized boolean isRunning() {
     return running;
   }
 
-  public synchronized void requestJenkinsSlave(Mesos.SlaveRequest request, Mesos.SlaveResult result) {
+  public synchronized void requestJenkinsSlave(Mesos.SlaveRequest slaveRequest, Mesos.SlaveResult slaveResult) {
     LOGGER.info("Enqueuing jenkins slave request");
-    requests.add(new Request(request, result));
+
+    Request request = new Request(slaveRequest, slaveResult);
+    if(isResourceLimitReached(request)) {
+      LOGGER.info("Maximum number of CPUs or Mem is reached, set request "+ slaveRequest.slave.name +" as failed " +
+              "for a later retry." );
+      slaveResult.failed(slaveRequest.slave, Mesos.SlaveResult.FAILED_CAUSE.RESOURCE_LIMIT_REACHED);
+      return;
+    }
+
+    requests.add(request);
+
     if (driver != null) {
       // Ask mesos to send all offers, even the those we declined earlier.
       // See comment in resourceOffers() for further details.
       driver.reviveOffers();
     }
   }
 
+  private boolean isResourceLimitReached(Request request) {
```

Comment:
make isCPULimitReached() and isMemoryLimitReached(), and use `return isCPULimitReached() || isMemoryLimitReached()`